src/Tasks/Common/FileUtilities.MetadataReader.cs

```diff
@@ -22,10 +22,15 @@ namespace Microsoft.NET.Build.Tasks
 {
     static partial class FileUtilities
     {
-        private static Dictionary<string, (DateTime LastKnownWriteTimeUtc, Version Version)> s_versionCache = new(StringComparer.OrdinalIgnoreCase /* Not strictly correct on *nix. Fix? */);
+        private static Dictionary<string, (DateTime LastKnownWriteTimeUtc, Version? Version)> s_versionCache = new(StringComparer.OrdinalIgnoreCase /* Not strictly correct on *nix. Fix? */);
 
-        private static Version GetAssemblyVersion(string sourcePath)
+        private static Version? GetAssemblyVersion(string? sourcePath)
         {
+            if (sourcePath == null)
```

Comment:
are empty paths valid here?

Reply:
So, the property that eventually flows to here could return null (from `ConflictItem.cs`):
![image](https://github.com/user-attachments/assets/952d5f3f-1bc0-4bfe-9453-850bb6e8c02f)

The value returned from this method ends up here, and you'll see that someone clearly didn't mind if it was `null` via the comment (also from `ConflictItem.cs`):
![image](https://github.com/user-attachments/assets/5bd55da2-6771-4930-be8c-c2d2e8d44d73)

Reply:
I've walked up that call stack I described above and changed the call to `TryGetAssemblyVersion` so that it won't use `string?` which then calls this `GetAssemblyVersion`. So now, this method no longer uses `string?`. So, this check is no longer here.